src/DeepCopy/Reflection/ReflectionHelper.php

```diff
@@ -27,8 +27,8 @@ public static function getProperties(\ReflectionClass $ref)
 
         if ($parentClass = $ref->getParentClass()) {
             $parentPropsArr = self::getProperties($parentClass);
-            foreach ($parentPropsArr as $property) {
-                $propsArr[] = $property;
```

Comment:
The `array_merge` also preserves the order of the vertical inheritance: parent class attributes don't overwrite current class ones.
The fix, even if is trivial, is not immediate: please create a PR with a test that try to covers every possible scenario, and only then commit the eventual fix.